packages/renderer/src/lib/dialogs/CommandPaletteUtils.ts

```diff
@@ -42,16 +43,104 @@ export function getGoToDisplayText(goToInfo: GoToInfo): string {
     return goToInfo.Name;
   } else if (goToInfo.type === 'Volume') {
     return goToInfo.Name.substring(0, 12);
+  } else if (goToInfo.type === 'Navigation') {
+    return goToInfo.name;
   }
   return 'Unknown';
 }
 
+// Helper function to extract and capitalize path prefix from link
+function extractPathPrefix(link: string, entryName: string): string | undefined {
+  // Remove leading slash and split by '/'
+  const pathSegments = link.replace(/^\//, '').split('/');
+
+  if (pathSegments.length === 0 || pathSegments[0] === '') {
+    return;
+  }
+
+  const firstSegment = pathSegments[0];
+
+  // For submenu items (like Kubernetes Dashboard), use the parent category
+  if (pathSegments.length > 1) {
+    const capitalizedSegment = firstSegment.charAt(0).toUpperCase() + firstSegment.slice(1);
```

Comment:
suggestion(non-blocking): use CSS to have the uppercase instead of javascript string manipulations